### PR TITLE
updates README docker instructions

### PR DESCRIPTION
**Pull request by p-ferreira**
- Updates docker instructions on README file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -283,26 +283,11 @@ A FRAME pallet is compromised of a number of blockchain primitives:
...
-# Run Substrate node without re-compiling
-./scripts/docker_run.sh ./target/release/node-template --dev --ws-external
-
-# Purge the local dev chain
-./scripts/docker_run.sh ./target/release/node-template purge-chain --dev
-
-# Check whether the code is compilable
-./scripts/docker_run.sh cargo check
+docker compose down --volumes
+sudo ./scripts/run/subtensor.sh -e docker --network mainnet --node-type lite
```
Comment (rajkaramchedu) [inline README.md:290]: So I believe we should add a `-b` flag explicitly. I have seen at least two cases where the users ran this command it came back with "node-subtensor command not found". This meant that the binary doesn't exist. Is this script supposed to build the "node-subtensor" without exception or only under some conditions? See a bit more detail in this Discord thread: https://discord.com/channels/799672011265015819/1176889736636407808/1219454860508794911

Comment (rajkaramchedu) [reply README.md:290]: Also, we really have to spell out on which OS versions this script is tested and validated. Right now this doc https://github.com/opentensor/subtensor/blob/main/docs/running-subtensor-locally.md and the docs.bittensor.com version https://docs.bittensor.com/getting-started/running-a-public-subtensor either don't state Docker support for which OSs or simply mislead by making the user infer that this works on macOS too. So, does the above script work on macOS or only on Linux. For Linux, which versions? arm64 or amd64 or both?

**Review by distributedstatemachine — COMMENTED** [review]
> Also, we really have to spell out on which OS versions this script is tested and validated. Right now this doc https://github.com/opentensor/subtensor/blob/main/docs/running-subtensor-locally.md and the docs.bittensor.com version https://docs.bittensor.com/getting-started/running-a-public-subtensor either don't state Docker support for which OSs or simply mislead by making the user infer that this works on macOS too. So, does the above script work on macOS or only on Linux. For Linux, which versions? arm64 or amd64 or both?  

Tested on OSX M3 Max.

The issue lies in using sudo on an M1. Users either need to enable running docker with sudo , or set `ENV CGO=1`. There is the issue that the images are currenly built for amd 

```
[+] Running 2/3btensor-mainnet-lite                                                                                          
 ✔ Network subtensor_default                                                                                                                                   Created0.0s  (linux/amd64) does not match the detected host platform (linux/arm64/v8) and n
[+] Running 2/3btensor-mainnet-lite                                                                                          
 ✔ Network subtensor_default                                                                                                                                   Created0.0s  (linux/amd64) does not match the detected host platform (linux/arm64/v8) and n
[+] Running 3/3btensor-mainnet-lite                                                                                          
 ✔ Network subtensor_default                                                                                                                                   Created0.0s  (linux/amd64) does not match the detected host platform (linux/arm64/v8) and n
 ✔ Container subtensor-mainnet-lite                                                                                                                            Started0.3s 
 ! mainnet-lite The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested 0.0s 
```

however , the node seems to be running fine. 


```
2024-03-22 05:54:59 Subtensor Node    
2024-03-22 05:54:59 ✌️  version 4.0.0-dev-unknown    
2024-03-22 05:54:59 ❤️  by Substrate DevHub <https://github.com/substrate-developer-hub>, 2017-2024    
2024-03-22 05:54:59 📋 Chain specification: Bittensor    
2024-03-22 05:54:59 🏷  Node name: illustrious-trains-1269    
2024-03-22 05:54:59 👤 Role: FULL    
2024-03-22 05:54:59 💾 Database: RocksDb at /tmp/blockchain/chains/bittensor/db/full    
2024-03-22 05:54:59 ⛓  Native runtime: node-subtensor-144 (node-subtensor-1.tx1.au1)    
2024-03-22 05:55:00 You're running on a system with a broken `madvise(MADV_DONTNEED)` implementation. This will result in lower performance.    
2024-03-22 05:55:04 🔨 Initializing Genesis block/state (state: 0x4015…9643, header-hash: 0x2f05…6c03)    
2024-03-22 05:55:06 🏷  Local node identity is: 12D3KooWSztoFHtkSFxkJTkPGk2gvQNvCiKQmkrdLKzqW5DT7u4a    
2024-03-22 05:55:07 💻 Operating system: linux    
2024-03-22 05:55:07 💻 CPU architecture: x86_64    
2024-03-22 05:55:07 💻 Target environment: gnu    
2024-03-22 05:55:07 💻 Memory: 63951MB    
2024-03-22 05:55:07 💻 Kernel: 5.15.49-linuxkit    
2024-03-22 05:55:07 💻 Linux distribution: Ubuntu 20.04.6 LTS    
2024-03-22 05:55:07 💻 Virtual machine: no    
2024-03-22 05:55:07 📦 Highest known block at #0    
2024-03-22 05:55:07 〽️ Prometheus exporter started at 127.0.0.1:9615    
2024-03-22 05:55:07 Running JSON-RPC HTTP server: addr=0.0.0.0:9933, allowed origins=["*"]    
2024-03-22 05:55:07 Running JSON-RPC WS server: addr=0.0.0.0:9944, allowed origins=["*"]    
2024-03-22 05:55:08 🔍 Discovered new external address for our node: /ip4/178.239.163.56/tcp/30333/ws/p2p/12D3KooWSztoFHtkSFxkJTkPGk2gvQNvCiKQmkrdLKzqW5DT7u4a    
2024-03-22 05:55:12 ⏩ Warping, Downloading state,
```

We should modify the build scripts to determine the arch , and use buildx to build for the approriate target. 

In addition

```
2024-03-22 05:55:00 You're running on a system with a broken `madvise(MADV_DONTNEED)` implementation. This will result in lower performance.
```

We might want to advise users to  enable rosetta, so docker does not fallback to  QEMU emulation: https://github.com/paritytech/polkadot/issues/7597#issuecomment-1677331834

**Comment by rajkaramchedu** [comment]
Love this digging into, thanks @samtvlabs. So @garrett-opentensor can you please take a look and see who can update and test the script with the above findings? Thanks!

**Comment by sam0x17** [comment]
just a general note, it's a pretty big anti-pattern to have to run docker with sudo so would be great if that could be avoided. The usual approach is to add the current user to the `docker` user group and re-start, or whatever.

**Comment by rajkaramchedu** [comment]
@sam0x17 Similar topic in this Issue: https://github.com/opentensor/subtensor/issues/298 for reference.

**Comment by distributedstatemachine** [comment]
@rajkaramchedu  is this something we can close now ? There has being a flurry of PRs addressing this , and wanted to be sure we have it covered

**Comment by rajkaramchedu** [comment]
Closing this. See #298.

**Pull request closed without merging**